photon-api/src/main/scala/com/linkedin/photon/ml/algorithm/RandomEffectCoordinate.scala

```diff
@@ -55,23 +56,24 @@ protected[ml] abstract class RandomEffectCoordinate[Objective <: SingleNodeObjec
 
   /**
    * Compute an optimized model (i.e. run the coordinate optimizer) for the current dataset using an existing model as
-   * a starting point
+   * a starting point.
    *
    * @param model The model to use as a starting point
    * @return A tuple of the updated model and the optimization states tracker
    */
   override protected[algorithm] def updateModel(
-    model: DatumScoringModel): (DatumScoringModel, Option[OptimizationTracker]) = model match {
-        case randomEffectModel: RandomEffectModel =>
-          RandomEffectCoordinate.updateModel(dataSet, optimizationProblem, randomEffectModel)
+      model: DatumScoringModel): (DatumScoringModel, Option[OptimizationTracker]) =
+    model match {
+      case randomEffectModel: RandomEffectModel =>
+        RandomEffectCoordinate.updateModel(dataSet, optimizationProblem, randomEffectModel)
 
-        case _ =>
-          throw new UnsupportedOperationException(s"Updating model of type ${model.getClass} " +
-              s"in ${this.getClass} is not supported!")
-      }
+      case _ =>
+        throw new UnsupportedOperationException(s"Updating model of type ${model.getClass} " +
+            s"in ${this.getClass} is not supported!")
```

Comment:
Exclamation mark can be removed. Error messages don't need, and mostly don't have, exclamation marks, IMHO.